test/e2e/proxy_test.go

```diff
@@ -15,25 +16,65 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/openshift/windows-machine-config-operator/controllers"
+	"github.com/openshift/windows-machine-config-operator/pkg/cluster"
 	"github.com/openshift/windows-machine-config-operator/pkg/patch"
 	"github.com/openshift/windows-machine-config-operator/pkg/retry"
+	"github.com/openshift/windows-machine-config-operator/pkg/windows"
 )
 
 // proxyTestSuite contains the validation cases for cluster-wide proxy.
 // All subtests are skipped if a proxy is not enabled in the test environment.
 func proxyTestSuite(t *testing.T) {
+	if !cluster.IsProxyEnabled() {
+		t.Skip("cluster-wide proxy is not enabled in this environment")
+	}
+
+	t.Run("Trusted CA ConfigMap validation", testTrustedCAConfigMap)
+	t.Run("Environment variables validation", testEnvVars)
+}
+
+// testEnvVars tests that on each node
+// 1. the system-level environment variables are set properly as per the cluster-wide proxy
+// 2. the required Windows services pick up the proper values for proxy environment variables
+func testEnvVars(t *testing.T) {
 	tc, err := NewTestContext()
 	require.NoError(t, err)
 
 	clusterProxy, err := tc.client.Config.ConfigV1().Proxies().Get(context.TODO(), "cluster", meta.GetOptions{})
 	if err != nil {
 		require.NoError(t, err)
 	}
-	if clusterProxy.Status.HTTPProxy == "" && clusterProxy.Status.HTTPSProxy == "" {
-		t.Skip("cluster-wide proxy is not enabled in this environment")
-	}
+	expectedEnvVars := map[string]string{}
+	expectedEnvVars["HTTP_PROXY"] = clusterProxy.Status.HTTPProxy
+	expectedEnvVars["HTTPS_PROXY"] = clusterProxy.Status.HTTPSProxy
+	expectedEnvVars["NO_PROXY"] = clusterProxy.Status.NoProxy
+	err = tc.waitForWindowsNodes(gc.numberOfMachineNodes, false, false, false)
```

Comment:
I suggest you use this instead https://github.com/openshift/windows-machine-config-operator/blob/f079f3ddbcbae368756f535ed78229295388924c/test/e2e/create_test.go#L564

Additionally, why is this only tested on Machine nodes?